fv3gfs/util/communicator.py

```diff
@@ -42,39 +43,76 @@ def wait(self):
         self._function()
 
 
+HaloSendTuple = Tuple[AsyncRequest, "Buffer"]
+HaloRequestSendList = List[HaloSendTuple]
+HaloRecvTuple = Tuple[AsyncRequest, "Buffer", ndarray]
+HaloRequestRecvList = List[HaloRecvTuple]
+
+
 class HaloUpdateRequest:
-    """asynchronous request object for halo updates"""
+    """Asynchronous request object for halo updates."""
+
+    _send_data = Union[HaloRequestSendList, List[Any]]
+    _recv_data = Union[HaloRequestRecvList, List[Any]]
```

Comment:
@mcgibbon I didn't want to put any but I couldn't figure out why I kept getting an assignment error with mypy with this `List[Any]` as the coming type. If you have an idea...

Reply:
You should be able to delete lines 55 and 56, there's no need to have these as declared class attributes. The type hint on init combined with the e.g. `self._send_data` assignment will automatically type-hint these attributes.